fem/nonlininteg.cpp

```diff
@@ -56,6 +56,58 @@ double NonlinearFormIntegrator::GetElementEnergy(
    return 0.0;
 }
 
+void BlockNonlinearFormIntegrator::AssembleElementVector(
```

Comment:
Why not make these methods pure virtual so that you don't have to write theses errors??

Reply:
I followed the pattern of the original nonlinearformintegrator class. I agree that pure virtual should catch if these methods are overloaded at compile-time. Any thoughts @tzanio and @v-dobrev ?

Reply:
The base integrator class `NonlinearFormIntegrator` defines two types of virtual methods that a derived class can implement: integration over the elements, and/or integration over the faces. With the default implementation in the base class, the derived classes need to implement only the one that makes sense for them.